src/components/Article/ArticlesList.jsx

```diff
@@ -1,18 +1,32 @@
 import { useEffect, useState } from 'react';
-import { getArticles } from '../../api';
+import { useSearchParams } from 'react-router-dom';
+import { getArticles, getArticlesByTopic } from '../../api';
 import ArticleStub from './ArticleStub';
 
 function ArticlesList({ articles, setArticles }) {
 
     const [isLoading, setIsLoading] = useState(true);
+    const [searchParams, setSearchParams] = useSearchParams();
 
     useEffect(()=> {
-        getArticles()
-        .then((fetchedArticles) => {
-            setArticles(fetchedArticles);
-            setIsLoading(false);
-        })
-    }, []);
+        if(searchParams.get("topic")){
+            getArticlesByTopic(searchParams.get("topic"))
+            .then((fetchedArticles) => {
+                setArticles(fetchedArticles);
+                setIsLoading(false);
+            })
+        }
+        else{
+            getArticles()
+            .then((fetchedArticles) => {
+                setArticles(fetchedArticles);
+                setIsLoading(false);
+            })
+        }
+
```

Comment:
After updating the api call function getArticles, this if statement will no longer needed. Can just call getArticles to deal with both all articles and filtered articles